src/jit/importer.cpp

```diff
@@ -3714,6 +3714,34 @@ GenTreePtr Compiler::impIntrinsic(GenTreePtr            newobjThis,
             break;
         }
 
+        case CORINFO_INTRINSIC_GetRawHandle:
+        {
+            assert(IsTargetAbi(CORINFO_CORERT_ABI)); // Only CoreRT supports it.
+            CORINFO_RESOLVED_TOKEN resolvedToken;
+            resolvedToken.hMethod      = method;
+            resolvedToken.tokenContext = MAKE_METHODCONTEXT(info.compMethodHnd);
+            resolvedToken.tokenScope   = info.compScopeHnd;
+            resolvedToken.token        = memberRef;
+            resolvedToken.tokenType    = CORINFO_TOKENKIND_Method;
+
+            CORINFO_GENERICHANDLE_RESULT embedInfo;
+            info.compCompHnd->expandRawHandleIntrinsic(&resolvedToken, &embedInfo);
+
+            GenTreePtr eeTypePtrOfNode = impLookupToTree(&resolvedToken, &embedInfo.lookup,
+                                                         gtTokenToIconFlags(memberRef), embedInfo.compileTimeHandle);
+            if (eeTypePtrOfNode == nullptr)
+                return nullptr;
+
+            unsigned eeSlot = lvaGrabTemp(true DEBUGARG("eeTypePtrOf"));
+            impAssignTempGen(eeSlot, eeTypePtrOfNode, clsHnd, (unsigned)CHECK_SPILL_NONE);
+            GenTreePtr lclVar     = gtNewLclvNode(eeSlot, TYP_I_IMPL);
+            GenTreePtr lclVarAddr = gtNewOperNode(GT_ADDR, TYP_I_IMPL, lclVar);
+            var_types  resultType = JITtype2varType(sig->retType);
```

Comment:
Would it be possible to add some kind of assert that the return type of the method is interchangeable with TYP_I_IMPL? (either an IntPtr, or a pointer-sized struct that is ideally not a GC pointer).